app/src/main/kotlin/com/hedvig/app/authenticate/LogoutUseCase.kt

```diff
@@ -1,52 +1,39 @@
 package com.hedvig.app.authenticate
 
 import com.apollographql.apollo3.ApolloClient
-import com.hedvig.android.apollo.OperationResult
-import com.hedvig.android.auth.AuthenticationTokenService
+import com.hedvig.android.auth.AuthTokenService
+import com.hedvig.android.core.common.ApplicationScope
+import com.hedvig.android.hanalytics.featureflags.FeatureManager
 import com.hedvig.android.market.MarketManager
 import com.hedvig.app.feature.chat.data.ChatEventStore
 import com.hedvig.app.feature.chat.data.UserRepository
 import com.hedvig.app.service.push.PushTokenManager
 import com.hedvig.app.util.apollo.reconnectSubscriptions
+import com.hedvig.hanalytics.HAnalytics
+import kotlinx.coroutines.launch
 
 class LogoutUseCase(
   private val pushTokenManager: PushTokenManager,
   private val marketManager: MarketManager,
-  private val loginStatusService: LoginStatusService,
   private val apolloClient: ApolloClient,
   private val userRepository: UserRepository,
-  private val authenticationTokenService: AuthenticationTokenService,
+  private val authTokenService: AuthTokenService,
   private val chatEventStore: ChatEventStore,
+  private val featureManager: FeatureManager,
+  private val hAnalytics: HAnalytics,
+  private val applicationScope: ApplicationScope,
 ) {
 
-  sealed class LogoutResult {
-    object Success : LogoutResult()
-    data class Error(val message: String?) : LogoutResult()
-  }
-
-  suspend fun invoke(): LogoutResult = when (val result = userRepository.logout()) {
-    is OperationResult.Error -> LogoutResult.Error(result.message)
-    is OperationResult.Success -> {
-      clearLoginStatus()
-      clearMarket()
-      clearAuthenticationToken()
-      apolloClient.reconnectSubscriptions()
-      runCatching { pushTokenManager.refreshToken() }
-      chatEventStore.resetChatClosedCounter()
-      LogoutResult.Success
+  fun invoke() {
+    applicationScope.launch { hAnalytics.loggedOut() }
+    applicationScope.launch { userRepository.logout() }
+    applicationScope.launch {
+      authTokenService.invalidateTokens()
     }
-  }
-
-  private fun clearAuthenticationToken() {
-    authenticationTokenService.authenticationToken = null
-  }
-
-  private fun clearMarket() {
-    marketManager.market = null
-  }
-
-  private fun clearLoginStatus() {
-    loginStatusService.isLoggedIn = false
-    authenticationTokenService.authenticationToken = null
+    applicationScope.launch { marketManager.market = null }
+    applicationScope.launch { pushTokenManager.refreshToken() }
+    applicationScope.launch { featureManager.invalidateExperiments() }
+    applicationScope.launch { chatEventStore.resetChatClosedCounter() }
+    applicationScope.launch { apolloClient.reconnectSubscriptions() }
```

Comment:
Launch them all individually like that since it's okay for them to fail individually if they ever do (not possible for most but still doesn't hurt to do so).